x-pack/test/security_solution_api_integration/test_suites/detections_response/rules_management/prebuilt_rules/management/trial_license_complete_tier/upgrade_review_prebuilt_rules.eql_query_fields.ts

```diff
@@ -0,0 +1,462 @@
+  const es = getService('es');
+  const supertest = getService('supertest');
+  const log = getService('log');
+
+  describe('@ess @serverless @skipInServerlessMKI review prebuilt rules updates from package with mock rule assets', () => {
+    beforeEach(async () => {
+      await deleteAllRules(supertest, log);
+      await deleteAllTimelines(es, log);
+      await deleteAllPrebuiltRuleAssets(es, log);
+    });
+
+    describe(`eql_query fields`, () => {
+      const getRuleAssetSavedObjects = () => [
+        createRuleAssetSavedObject({
+          rule_id: 'rule-1',
+          version: 1,
+          type: 'eql',
+          query: 'query where true',
+          language: 'eql',
+          filters: [],
+        }),
+      ];
+
+      describe("when rule field doesn't have an update and has no custom value - scenario AAA", () => {
+        it('should not show in the upgrade/_review API response', async () => {
+          // Install base prebuilt detection rule
+          await createHistoricalPrebuiltRuleAssetSavedObjects(es, getRuleAssetSavedObjects());
+          await installPrebuiltRules(es, supertest);
+
+          // Increment the version of the installed rule, do NOT update the related eql_query field, and create the new rule assets
+          const updatedRuleAssetSavedObjects = [
+            createRuleAssetSavedObject({
+              rule_id: 'rule-1',
+              version: 2,
+              type: 'eql',
+              query: 'query where true',
+              language: 'eql',
+              filters: [],
+            }),
+          ];
+          await createHistoricalPrebuiltRuleAssetSavedObjects(es, updatedRuleAssetSavedObjects);
+
+          // Call the upgrade review prebuilt rules endpoint and check that there is 1 rule eligable for update but eql_query field is NOT returned
+          const reviewResponse = await reviewPrebuiltRulesToUpgrade(supertest);
+          const fieldDiffObject = reviewResponse.rules[0].diff.fields as AllFieldsDiff;
+          expect(fieldDiffObject.eql_query).toBeUndefined();
+
+          expect(reviewResponse.rules[0].diff.num_fields_with_updates).toBe(1);
+          expect(reviewResponse.rules[0].diff.num_fields_with_conflicts).toBe(0);
+          expect(reviewResponse.rules[0].diff.num_fields_with_non_solvable_conflicts).toBe(0);
+          expect(reviewResponse.stats.num_rules_with_conflicts).toBe(0);
+          expect(reviewResponse.stats.num_rules_with_non_solvable_conflicts).toBe(0);
+        });
+      });
+
+      describe("when rule field doesn't have an update but has a custom value - scenario ABA", () => {
+        it('should show in the upgrade/_review API response', async () => {
+          // Install base prebuilt detection rule
+          await createHistoricalPrebuiltRuleAssetSavedObjects(es, getRuleAssetSavedObjects());
+          await installPrebuiltRules(es, supertest);
+
+          // Customize an eql_query field on the installed rule
+          await updateRule(supertest, {
```

Comment:
No, this is copied over from the `kql_query` file where the difference was needed, I can switch them to be the same

Reply:
nit: I think it's better to use `patchRule` because it communicates the intent more clearly. Like "I want to update these specific fields without touching the other ones".

Reply:
I'd probably agree with you for the ones that don't need to unset fields, but for whatever reason, certain fields like `saved_id` and `threat_query` don't correctly update with this `PATCH` util. I didn't look too much into it as I thought keeping the tests consistent and readable was a good reason to just use the `updateRule` util, possible I was just using it incorrectly. But it works with other fields so not sure, I'll look more into it soon and open a ticket to investigate if need be